### PR TITLE
Refactor/apply vitepress templates

### PR DESCRIPTION
**Pull request by neodmy**
- Bump Vitepress
- Split index.md into several files for the features section
- Add team.md
- Add sidebar config

### CODE REVIEW

**Review by inigomarquinez — CHANGES_REQUESTED** [review]
Great improvement, @neodmy !

I've added a comment and apart from that I have a question. Do we need to upload the vitepress cache folder to GitHub? Or is it something that is generated when building the app?.

File: docs/.vitepress/config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  title: '12-factor App Pills',
+  title: 'The Twelve-Factor App Pills',
```
[the remark this answers is not in the document]
Comment (neodmy) [reply docs/.vitepress/config.js:2]: Done

**Comment by neodmy** [comment]
> Great improvement, @neodmy !
> 
> I've added a comment and apart from that I have a question. Do we need to upload the vitepress cache folder to GitHub? Or is it something that is generated when building the app?.

I've removed the cache folder

**Review by inigomarquinez — APPROVED** [review]
LGTM!

**Pull request merged**